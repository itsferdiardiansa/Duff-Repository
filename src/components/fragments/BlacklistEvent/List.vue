<template>
  <div class="container">
    <div class="wrapper">
      <Table
        :headers="tHeaders"
        :items="filteredData"
        :showLoader="isFetching"
        :rowLoader="2"
      >
        <template v-slot:blacklist="props">
          <div>
            <h3 class="font-black textr-lg mb-2">{{ props.data.title }}</h3>

            <p>
              blacklist at: <span>{{ props.data.blacklistAt }}</span>
            </p>
            <p>
              reported at: <span>{{ props.data.reportedAt }}</span>
            </p>
          </div>
        </template>

        <template v-slot:action>
          <Button :textBold="true" size="sm" variant="warning">
            <img svg-inline class="w-4 h-4" src="@icon/pencil.svg" />
          </Button>
        </template>
      </Table>
    </div>
  </div>
</template>
<script>
import { computed, ref } from 'vue'
import { useStore } from 'vuex'
import Table from '@common/Table'
import Button from '@common/Button'
import data from '@mock/collections'

export default {
  components: {
    Button,
    Table,
  },
  setup() {
    const store = useStore()

    let tHeaders = ref([
      {
        title: 'Blacklist Event Name / Desc / TNC',
        accessor: 'blacklist',
        width: '40%',
        align: 'left',
      },
      {
        title: 'Created Date',
        accessor: 'createdDate',
        width: '60%',
        align: 'left',
      },
      {
        title: 'Action',
        accessor: 'action',
        colSpan: 3,
        width: '5%',
        align: 'center',
      },
    ])

    const filteredData = computed(() => {
      return data.blacklistEvent.result
    })

    const isFetching = computed(() => {
      return store.getters['thematicPage/getFetchStatus']
    })

    return {
      isFetching,
      filteredData,
      tHeaders,
    }
  },
}
</script>
<style lang="scss" scoped>
.container {
  @apply relative;

  .wrapper {
    @apply relative;
  }
}
</style>
