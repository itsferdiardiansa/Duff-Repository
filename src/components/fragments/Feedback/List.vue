<template>
  <div class="list">
    <Table
      :headers="tHeaders"
      :items="filteredData"
      :isFetching="requestStatus.fetch"
      :onError="requestStatus.error.status"
      :pagination="pagination"
      :onPageChange="handlePageChange"
      :onFailedFetchHandler="fetchData"
    >
      <template #read_by_slice="{ data }">
        <div class="read-by-col">
          <template v-for="(item, key) in data.read_by_slice" :key="key">
            <Badge variant="primary" :label="item" />
          </template>
        </div>
      </template>

      <template #responded_by="{ data }">
        <Badge variant="primary" :label="data.responded_by" />
      </template>

      <template #action="{ data }">
        <ActionButton :data="actionButtons" :item="data" />
      </template>
    </Table>
  </div>
</template>
<script>
import { computed, onMounted, reactive, ref } from 'vue';
import { useStore } from 'vuex';
import Table, { ActionButton } from '@common/Table';
import Badge from '@common/Badge';

export default {
  components: {
    Table,
    Badge,
    ActionButton,
  },
  setup() {
    const store = useStore();
    const params = reactive({ page: 1, limit: 5 });

    const tHeaders = ref([
      {
        title: 'Email',
        accessor: 'email',
        width: '15%',
        align: 'left',
      },
      { title: 'Name', accessor: 'name', width: '15%' },
      { title: 'Page', accessor: 'page' },
      { title: 'Section', accessor: 'section' },
      { title: 'Responded by', accessor: 'responded_by' },
      { title: 'Read by', accessor: 'read_by_slice' },
      {
        title: 'Action',
        accessor: 'action',
        colSpan: 1,
        align: 'center',
      },
    ]);

    const actionButtons = reactive([
      {
        variant: 'dark',
        icon: ['fa', 'marker'],
        text: 'Mark as responded',
        bold: true,
        onClickFn: (e, data) => {
          markAsResponded(e, data);
        },
      },
      {
        variant: 'dark',
        icon: ['fa', 'bookmark'],
        text: 'Mark as read',
        inverse: true,
        bold: true,
        onClickFn: (e, data) => {
          markAsRead(e, data);
        },
      },
    ]);

    const fetchData = () => {
      store.dispatch('feedback/fetchData', params);
    };

    const markAsResponded = (e, data) => {
      store.dispatch('feedback/markAsResponded', {
        action: 'form.create',
        data,
        params,
      });
    };

    const markAsRead = (e, data) => {
      store.dispatch('feedback/markAsRead', {
        action: 'form.create',
        data,
        params,
      });
    };

    const filteredData = computed(() => {
      return store.getters['feedback/getItems'];
    });

    const pagination = computed(() => {
      return store.getters['feedback/getPagination'];
    });

    const requestStatus = computed(() => {
      return store.getters['feedback/getRequestStatus'];
    });

    const handlePageChange = pageParams => {
      Object.assign(params, pageParams);

      fetchData();
    };

    onMounted(fetchData);

    return {
      fetchData,
      requestStatus,
      filteredData,
      tHeaders,
      actionButtons,
      pagination,
      handlePageChange,
    };
  },
};
</script>
<style lang="scss" scoped>
.list {
  @apply relative;

  .wrapper {
    @apply relative;
  }
  .read-by-col {
    @apply flex flex-wrap justify-center;
    max-width: 200px;

    span {
      @apply mt-1 mr-1;
    }
  }
}
</style>
