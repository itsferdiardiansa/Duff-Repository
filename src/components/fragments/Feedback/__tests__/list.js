import { mount } from '@vue/test-utils';
import { createStore } from 'vuex';
import List from '@fragment/Feedback/List';
import feedbackModule from '@store/modules/Feedback';

describe('fragment/Feedback/List', () => {
  let store;

  beforeAll(() => {
    store = createStore({
      modules: {
        feedback: feedbackModule,
      },
    });
  });

  it('will render and match snapshot', async () => {
    const Component = mount(<List />, {
      global: {
        plugins: [store],
      },
    });

    await Component.vm.$store.dispatch('feedback/fetchData', {
      page: 1,
      limit: 10,
    });
    expect(Component.html()).toMatchSnapshot();
  });
});
