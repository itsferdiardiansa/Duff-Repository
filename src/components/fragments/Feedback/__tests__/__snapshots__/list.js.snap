// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fragment/Feedback/List will render and match snapshot 1`] = `
<div class="list">
  <!--v-if-->
  <table class="ss-table">
    <thead class="ss-table--head" headers="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]" items="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]" isfetching="false" pagination="[object Object]" emptymessage="Empty data" rownumber="true" rowloader="5" selectablerows="false" withpagination="true" withfilter="false" filteroptions="">
      <tr>
        <th width="4%">
          <!--v-if--><span>#</span>
        </th>
        <th class="" width="15%" align="left">Email</th>
        <th class="" width="15%">Name</th>
        <th class="">Page</th>
        <th class="">Section</th>
        <th class="">Responded by</th>
        <th class="">Read by</th>
        <th colspan="1" class="" align="center">Action</th>
      </tr>
    </thead>
    <tbody class="ss-table--body">
      <!--v-if-->
      <tr class="table-content--body-row">
        <td class="ss-table--body-col"><label>1</label></td>
        <td class="ss-table--body-col text-left">
          <div>test01@test.com</div>
        </td>
        <td class="ss-table--body-col">
          <div>dev test</div>
        </td>
        <td class="ss-table--body-col">
          <div>create-event</div>
        </td>
        <td class="ss-table--body-col">
          <div></div>
        </td>
        <td class="ss-table--body-col"><span class="ss-badge ss-badge--primary ss-badge--xs text-bold"><div class="ss-badge--wrapper"><!--v-if--><div class="ss-badge--label">qa@loket.com</div></div></span></td>
        <td class="ss-table--body-col">
          <div class="read-by-col"><span class="ss-badge ss-badge--primary ss-badge--xs text-bold"><div class="ss-badge--wrapper"><!--v-if--><div class="ss-badge--label">qa@loket.com</div></div></span></div>
        </td>
        <td class="ss-table--body-col text-center">
          <div class="ss-tb--action">
            <div class="wrapper"><button class="ss-button ss-button--dark ss-button--base text-bold" data-id="1614839174743">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 512 512">
                      <path fill="currentColor" d="M93.95 290.03A327.038 327.038 0 0 0 .17 485.11l-.03.23c-1.7 15.28 11.21 28.2 26.49 26.51a327.02 327.02 0 0 0 195.34-93.8l75.4-75.4-128.02-128.02-75.4 75.4zM485.49 26.51c-35.35-35.35-92.67-35.35-128.02 0l-21.76 21.76-36.56-36.55c-15.62-15.62-40.95-15.62-56.56 0L138.47 115.84c-6.25 6.25-6.25 16.38 0 22.63l22.62 22.62c6.25 6.25 16.38 6.25 22.63 0l87.15-87.15 19.59 19.59L191.98 192 320 320.02l165.49-165.49c35.35-35.35 35.35-92.66 0-128.02z"></path>
                    </svg></div>
                  <div class="ss-button--label has-icon">Mark as responded</div>
                </div>
              </button><button class="ss-button ss-button--dark-inverse ss-button--base text-bold" data-id="1614839174743">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 384 512">
                      <path fill="currentColor" d="M0 512V48C0 21.49 21.49 0 48 0h288c26.51 0 48 21.49 48 48v464L192 400 0 512z"></path>
                    </svg></div>
                  <div class="ss-button--label has-icon">Mark as read</div>
                </div>
              </button></div>
          </div>
        </td>
      </tr>
      <tr class="table-content--body-row">
        <td class="ss-table--body-col"><label>2</label></td>
        <td class="ss-table--body-col text-left">
          <div>test@test.com</div>
        </td>
        <td class="ss-table--body-col">
          <div>firstname lastname</div>
        </td>
        <td class="ss-table--body-col">
          <div>create-event</div>
        </td>
        <td class="ss-table--body-col">
          <div></div>
        </td>
        <td class="ss-table--body-col"><span class="ss-badge ss-badge--primary ss-badge--xs text-bold"><div class="ss-badge--wrapper"><!--v-if--><div class="ss-badge--label">qa@loket.com</div></div></span></td>
        <td class="ss-table--body-col">
          <div class="read-by-col"><span class="ss-badge ss-badge--primary ss-badge--xs text-bold"><div class="ss-badge--wrapper"><!--v-if--><div class="ss-badge--label">qa@loket.com</div></div></span></div>
        </td>
        <td class="ss-table--body-col text-center">
          <div class="ss-tb--action">
            <div class="wrapper"><button class="ss-button ss-button--dark ss-button--base text-bold" data-id="1614839174759">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 512 512">
                      <path fill="currentColor" d="M93.95 290.03A327.038 327.038 0 0 0 .17 485.11l-.03.23c-1.7 15.28 11.21 28.2 26.49 26.51a327.02 327.02 0 0 0 195.34-93.8l75.4-75.4-128.02-128.02-75.4 75.4zM485.49 26.51c-35.35-35.35-92.67-35.35-128.02 0l-21.76 21.76-36.56-36.55c-15.62-15.62-40.95-15.62-56.56 0L138.47 115.84c-6.25 6.25-6.25 16.38 0 22.63l22.62 22.62c6.25 6.25 16.38 6.25 22.63 0l87.15-87.15 19.59 19.59L191.98 192 320 320.02l165.49-165.49c35.35-35.35 35.35-92.66 0-128.02z"></path>
                    </svg></div>
                  <div class="ss-button--label has-icon">Mark as responded</div>
                </div>
              </button><button class="ss-button ss-button--dark-inverse ss-button--base text-bold" data-id="1614839174759">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 384 512">
                      <path fill="currentColor" d="M0 512V48C0 21.49 21.49 0 48 0h288c26.51 0 48 21.49 48 48v464L192 400 0 512z"></path>
                    </svg></div>
                  <div class="ss-button--label has-icon">Mark as read</div>
                </div>
              </button></div>
          </div>
        </td>
      </tr>
      <tr class="table-content--body-row">
        <td class="ss-table--body-col"><label>3</label></td>
        <td class="ss-table--body-col text-left">
          <div>test@test.com</div>
        </td>
        <td class="ss-table--body-col">
          <div>firstname lastname</div>
        </td>
        <td class="ss-table--body-col">
          <div>create-event</div>
        </td>
        <td class="ss-table--body-col">
          <div></div>
        </td>
        <td class="ss-table--body-col"><span class="ss-badge ss-badge--primary ss-badge--xs text-bold"><div class="ss-badge--wrapper"><!--v-if--><div class="ss-badge--label">qa@loket.com</div></div></span></td>
        <td class="ss-table--body-col">
          <div class="read-by-col"><span class="ss-badge ss-badge--primary ss-badge--xs text-bold"><div class="ss-badge--wrapper"><!--v-if--><div class="ss-badge--label">qa@loket.com</div></div></span></div>
        </td>
        <td class="ss-table--body-col text-center">
          <div class="ss-tb--action">
            <div class="wrapper"><button class="ss-button ss-button--dark ss-button--base text-bold" data-id="1614839174771">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 512 512">
                      <path fill="currentColor" d="M93.95 290.03A327.038 327.038 0 0 0 .17 485.11l-.03.23c-1.7 15.28 11.21 28.2 26.49 26.51a327.02 327.02 0 0 0 195.34-93.8l75.4-75.4-128.02-128.02-75.4 75.4zM485.49 26.51c-35.35-35.35-92.67-35.35-128.02 0l-21.76 21.76-36.56-36.55c-15.62-15.62-40.95-15.62-56.56 0L138.47 115.84c-6.25 6.25-6.25 16.38 0 22.63l22.62 22.62c6.25 6.25 16.38 6.25 22.63 0l87.15-87.15 19.59 19.59L191.98 192 320 320.02l165.49-165.49c35.35-35.35 35.35-92.66 0-128.02z"></path>
                    </svg></div>
                  <div class="ss-button--label has-icon">Mark as responded</div>
                </div>
              </button><button class="ss-button ss-button--dark-inverse ss-button--base text-bold" data-id="1614839174771">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 384 512">
                      <path fill="currentColor" d="M0 512V48C0 21.49 21.49 0 48 0h288c26.51 0 48 21.49 48 48v464L192 400 0 512z"></path>
                    </svg></div>
                  <div class="ss-button--label has-icon">Mark as read</div>
                </div>
              </button></div>
          </div>
        </td>
      </tr>
      <tr class="table-content--body-row">
        <td class="ss-table--body-col"><label>4</label></td>
        <td class="ss-table--body-col text-left">
          <div>test@test01.com</div>
        </td>
        <td class="ss-table--body-col">
          <div>firstname lastname</div>
        </td>
        <td class="ss-table--body-col">
          <div>create-event</div>
        </td>
        <td class="ss-table--body-col">
          <div></div>
        </td>
        <td class="ss-table--body-col"><span class="ss-badge ss-badge--primary ss-badge--xs text-bold"><div class="ss-badge--wrapper"><!--v-if--><div class="ss-badge--label">qa@loket.com</div></div></span></td>
        <td class="ss-table--body-col">
          <div class="read-by-col"><span class="ss-badge ss-badge--primary ss-badge--xs text-bold"><div class="ss-badge--wrapper"><!--v-if--><div class="ss-badge--label">qa@loket.com</div></div></span></div>
        </td>
        <td class="ss-table--body-col text-center">
          <div class="ss-tb--action">
            <div class="wrapper"><button class="ss-button ss-button--dark ss-button--base text-bold" data-id="1614839174778">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 512 512">
                      <path fill="currentColor" d="M93.95 290.03A327.038 327.038 0 0 0 .17 485.11l-.03.23c-1.7 15.28 11.21 28.2 26.49 26.51a327.02 327.02 0 0 0 195.34-93.8l75.4-75.4-128.02-128.02-75.4 75.4zM485.49 26.51c-35.35-35.35-92.67-35.35-128.02 0l-21.76 21.76-36.56-36.55c-15.62-15.62-40.95-15.62-56.56 0L138.47 115.84c-6.25 6.25-6.25 16.38 0 22.63l22.62 22.62c6.25 6.25 16.38 6.25 22.63 0l87.15-87.15 19.59 19.59L191.98 192 320 320.02l165.49-165.49c35.35-35.35 35.35-92.66 0-128.02z"></path>
                    </svg></div>
                  <div class="ss-button--label has-icon">Mark as responded</div>
                </div>
              </button><button class="ss-button ss-button--dark-inverse ss-button--base text-bold" data-id="1614839174778">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 384 512">
                      <path fill="currentColor" d="M0 512V48C0 21.49 21.49 0 48 0h288c26.51 0 48 21.49 48 48v464L192 400 0 512z"></path>
                    </svg></div>
                  <div class="ss-button--label has-icon">Mark as read</div>
                </div>
              </button></div>
          </div>
        </td>
      </tr>
      <tr class="table-content--body-row">
        <td class="ss-table--body-col"><label>5</label></td>
        <td class="ss-table--body-col text-left">
          <div>test@test01.com</div>
        </td>
        <td class="ss-table--body-col">
          <div>firstname lastname</div>
        </td>
        <td class="ss-table--body-col">
          <div>create-event</div>
        </td>
        <td class="ss-table--body-col">
          <div></div>
        </td>
        <td class="ss-table--body-col"><span class="ss-badge ss-badge--primary ss-badge--xs text-bold"><div class="ss-badge--wrapper"><!--v-if--><div class="ss-badge--label">qa@loket.com</div></div></span></td>
        <td class="ss-table--body-col">
          <div class="read-by-col"><span class="ss-badge ss-badge--primary ss-badge--xs text-bold"><div class="ss-badge--wrapper"><!--v-if--><div class="ss-badge--label">qa@loket.com</div></div></span></div>
        </td>
        <td class="ss-table--body-col text-center">
          <div class="ss-tb--action">
            <div class="wrapper"><button class="ss-button ss-button--dark ss-button--base text-bold" data-id="1614839174788">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 512 512">
                      <path fill="currentColor" d="M93.95 290.03A327.038 327.038 0 0 0 .17 485.11l-.03.23c-1.7 15.28 11.21 28.2 26.49 26.51a327.02 327.02 0 0 0 195.34-93.8l75.4-75.4-128.02-128.02-75.4 75.4zM485.49 26.51c-35.35-35.35-92.67-35.35-128.02 0l-21.76 21.76-36.56-36.55c-15.62-15.62-40.95-15.62-56.56 0L138.47 115.84c-6.25 6.25-6.25 16.38 0 22.63l22.62 22.62c6.25 6.25 16.38 6.25 22.63 0l87.15-87.15 19.59 19.59L191.98 192 320 320.02l165.49-165.49c35.35-35.35 35.35-92.66 0-128.02z"></path>
                    </svg></div>
                  <div class="ss-button--label has-icon">Mark as responded</div>
                </div>
              </button><button class="ss-button ss-button--dark-inverse ss-button--base text-bold" data-id="1614839174788">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 384 512">
                      <path fill="currentColor" d="M0 512V48C0 21.49 21.49 0 48 0h288c26.51 0 48 21.49 48 48v464L192 400 0 512z"></path>
                    </svg></div>
                  <div class="ss-button--label has-icon">Mark as read</div>
                </div>
              </button></div>
          </div>
        </td>
      </tr>
      <tr class="table-content--body-row">
        <td class="ss-table--body-col"><label>6</label></td>
        <td class="ss-table--body-col text-left">
          <div>test@test.com</div>
        </td>
        <td class="ss-table--body-col">
          <div>firstname lastname</div>
        </td>
        <td class="ss-table--body-col">
          <div>create-event</div>
        </td>
        <td class="ss-table--body-col">
          <div></div>
        </td>
        <td class="ss-table--body-col"><span class="ss-badge ss-badge--primary ss-badge--xs text-bold"><div class="ss-badge--wrapper"><!--v-if--><div class="ss-badge--label">qa@loket.com</div></div></span></td>
        <td class="ss-table--body-col">
          <div class="read-by-col"><span class="ss-badge ss-badge--primary ss-badge--xs text-bold"><div class="ss-badge--wrapper"><!--v-if--><div class="ss-badge--label">qa@loket.com</div></div></span></div>
        </td>
        <td class="ss-table--body-col text-center">
          <div class="ss-tb--action">
            <div class="wrapper"><button class="ss-button ss-button--dark ss-button--base text-bold" data-id="1614839174798">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 512 512">
                      <path fill="currentColor" d="M93.95 290.03A327.038 327.038 0 0 0 .17 485.11l-.03.23c-1.7 15.28 11.21 28.2 26.49 26.51a327.02 327.02 0 0 0 195.34-93.8l75.4-75.4-128.02-128.02-75.4 75.4zM485.49 26.51c-35.35-35.35-92.67-35.35-128.02 0l-21.76 21.76-36.56-36.55c-15.62-15.62-40.95-15.62-56.56 0L138.47 115.84c-6.25 6.25-6.25 16.38 0 22.63l22.62 22.62c6.25 6.25 16.38 6.25 22.63 0l87.15-87.15 19.59 19.59L191.98 192 320 320.02l165.49-165.49c35.35-35.35 35.35-92.66 0-128.02z"></path>
                    </svg></div>
                  <div class="ss-button--label has-icon">Mark as responded</div>
                </div>
              </button><button class="ss-button ss-button--dark-inverse ss-button--base text-bold" data-id="1614839174798">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 384 512">
                      <path fill="currentColor" d="M0 512V48C0 21.49 21.49 0 48 0h288c26.51 0 48 21.49 48 48v464L192 400 0 512z"></path>
                    </svg></div>
                  <div class="ss-button--label has-icon">Mark as read</div>
                </div>
              </button></div>
          </div>
        </td>
      </tr>
      <tr class="table-content--body-row">
        <td class="ss-table--body-col"><label>7</label></td>
        <td class="ss-table--body-col text-left">
          <div>test@test.com</div>
        </td>
        <td class="ss-table--body-col">
          <div>firstname lastname</div>
        </td>
        <td class="ss-table--body-col">
          <div>create-event</div>
        </td>
        <td class="ss-table--body-col">
          <div></div>
        </td>
        <td class="ss-table--body-col"><span class="ss-badge ss-badge--primary ss-badge--xs text-bold"><div class="ss-badge--wrapper"><!--v-if--><div class="ss-badge--label">qa@loket.com</div></div></span></td>
        <td class="ss-table--body-col">
          <div class="read-by-col"><span class="ss-badge ss-badge--primary ss-badge--xs text-bold"><div class="ss-badge--wrapper"><!--v-if--><div class="ss-badge--label">qa@loket.com</div></div></span></div>
        </td>
        <td class="ss-table--body-col text-center">
          <div class="ss-tb--action">
            <div class="wrapper"><button class="ss-button ss-button--dark ss-button--base text-bold" data-id="1614839174828">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 512 512">
                      <path fill="currentColor" d="M93.95 290.03A327.038 327.038 0 0 0 .17 485.11l-.03.23c-1.7 15.28 11.21 28.2 26.49 26.51a327.02 327.02 0 0 0 195.34-93.8l75.4-75.4-128.02-128.02-75.4 75.4zM485.49 26.51c-35.35-35.35-92.67-35.35-128.02 0l-21.76 21.76-36.56-36.55c-15.62-15.62-40.95-15.62-56.56 0L138.47 115.84c-6.25 6.25-6.25 16.38 0 22.63l22.62 22.62c6.25 6.25 16.38 6.25 22.63 0l87.15-87.15 19.59 19.59L191.98 192 320 320.02l165.49-165.49c35.35-35.35 35.35-92.66 0-128.02z"></path>
                    </svg></div>
                  <div class="ss-button--label has-icon">Mark as responded</div>
                </div>
              </button><button class="ss-button ss-button--dark-inverse ss-button--base text-bold" data-id="1614839174828">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 384 512">
                      <path fill="currentColor" d="M0 512V48C0 21.49 21.49 0 48 0h288c26.51 0 48 21.49 48 48v464L192 400 0 512z"></path>
                    </svg></div>
                  <div class="ss-button--label has-icon">Mark as read</div>
                </div>
              </button></div>
          </div>
        </td>
      </tr>
      <tr class="table-content--body-row">
        <td class="ss-table--body-col"><label>8</label></td>
        <td class="ss-table--body-col text-left">
          <div>test@test.com</div>
        </td>
        <td class="ss-table--body-col">
          <div>firstname lastname</div>
        </td>
        <td class="ss-table--body-col">
          <div>create-event</div>
        </td>
        <td class="ss-table--body-col">
          <div></div>
        </td>
        <td class="ss-table--body-col"><span class="ss-badge ss-badge--primary ss-badge--xs text-bold"><div class="ss-badge--wrapper"><!--v-if--><div class="ss-badge--label">qa@loket.com</div></div></span></td>
        <td class="ss-table--body-col">
          <div class="read-by-col"><span class="ss-badge ss-badge--primary ss-badge--xs text-bold"><div class="ss-badge--wrapper"><!--v-if--><div class="ss-badge--label">qa@loket.com</div></div></span></div>
        </td>
        <td class="ss-table--body-col text-center">
          <div class="ss-tb--action">
            <div class="wrapper"><button class="ss-button ss-button--dark ss-button--base text-bold" data-id="1614839174896">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 512 512">
                      <path fill="currentColor" d="M93.95 290.03A327.038 327.038 0 0 0 .17 485.11l-.03.23c-1.7 15.28 11.21 28.2 26.49 26.51a327.02 327.02 0 0 0 195.34-93.8l75.4-75.4-128.02-128.02-75.4 75.4zM485.49 26.51c-35.35-35.35-92.67-35.35-128.02 0l-21.76 21.76-36.56-36.55c-15.62-15.62-40.95-15.62-56.56 0L138.47 115.84c-6.25 6.25-6.25 16.38 0 22.63l22.62 22.62c6.25 6.25 16.38 6.25 22.63 0l87.15-87.15 19.59 19.59L191.98 192 320 320.02l165.49-165.49c35.35-35.35 35.35-92.66 0-128.02z"></path>
                    </svg></div>
                  <div class="ss-button--label has-icon">Mark as responded</div>
                </div>
              </button><button class="ss-button ss-button--dark-inverse ss-button--base text-bold" data-id="1614839174896">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 384 512">
                      <path fill="currentColor" d="M0 512V48C0 21.49 21.49 0 48 0h288c26.51 0 48 21.49 48 48v464L192 400 0 512z"></path>
                    </svg></div>
                  <div class="ss-button--label has-icon">Mark as read</div>
                </div>
              </button></div>
          </div>
        </td>
      </tr>
      <tr class="table-content--body-row">
        <td class="ss-table--body-col"><label>9</label></td>
        <td class="ss-table--body-col text-left">
          <div>test@test.com</div>
        </td>
        <td class="ss-table--body-col">
          <div>firstname lastname</div>
        </td>
        <td class="ss-table--body-col">
          <div>create-event</div>
        </td>
        <td class="ss-table--body-col">
          <div></div>
        </td>
        <td class="ss-table--body-col"><span class="ss-badge ss-badge--primary ss-badge--xs text-bold"><div class="ss-badge--wrapper"><!--v-if--><div class="ss-badge--label">qa@loket.com</div></div></span></td>
        <td class="ss-table--body-col">
          <div class="read-by-col"><span class="ss-badge ss-badge--primary ss-badge--xs text-bold"><div class="ss-badge--wrapper"><!--v-if--><div class="ss-badge--label">qa@loket.com</div></div></span></div>
        </td>
        <td class="ss-table--body-col text-center">
          <div class="ss-tb--action">
            <div class="wrapper"><button class="ss-button ss-button--dark ss-button--base text-bold" data-id="1614839174930">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 512 512">
                      <path fill="currentColor" d="M93.95 290.03A327.038 327.038 0 0 0 .17 485.11l-.03.23c-1.7 15.28 11.21 28.2 26.49 26.51a327.02 327.02 0 0 0 195.34-93.8l75.4-75.4-128.02-128.02-75.4 75.4zM485.49 26.51c-35.35-35.35-92.67-35.35-128.02 0l-21.76 21.76-36.56-36.55c-15.62-15.62-40.95-15.62-56.56 0L138.47 115.84c-6.25 6.25-6.25 16.38 0 22.63l22.62 22.62c6.25 6.25 16.38 6.25 22.63 0l87.15-87.15 19.59 19.59L191.98 192 320 320.02l165.49-165.49c35.35-35.35 35.35-92.66 0-128.02z"></path>
                    </svg></div>
                  <div class="ss-button--label has-icon">Mark as responded</div>
                </div>
              </button><button class="ss-button ss-button--dark-inverse ss-button--base text-bold" data-id="1614839174930">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 384 512">
                      <path fill="currentColor" d="M0 512V48C0 21.49 21.49 0 48 0h288c26.51 0 48 21.49 48 48v464L192 400 0 512z"></path>
                    </svg></div>
                  <div class="ss-button--label has-icon">Mark as read</div>
                </div>
              </button></div>
          </div>
        </td>
      </tr>
      <tr class="table-content--body-row">
        <td class="ss-table--body-col"><label>10</label></td>
        <td class="ss-table--body-col text-left">
          <div>ferdi.ardiansa@gojek.com</div>
        </td>
        <td class="ss-table--body-col">
          <div>Random</div>
        </td>
        <td class="ss-table--body-col">
          <div>Dashboard</div>
        </td>
        <td class="ss-table--body-col">
          <div>General</div>
        </td>
        <td class="ss-table--body-col"><span class="ss-badge ss-badge--primary ss-badge--xs text-bold"><div class="ss-badge--wrapper"><!--v-if--><div class="ss-badge--label">qa@loket.com</div></div></span></td>
        <td class="ss-table--body-col">
          <div class="read-by-col"><span class="ss-badge ss-badge--primary ss-badge--xs text-bold"><div class="ss-badge--wrapper"><!--v-if--><div class="ss-badge--label">qa@loket.com</div></div></span></div>
        </td>
        <td class="ss-table--body-col text-center">
          <div class="ss-tb--action">
            <div class="wrapper"><button class="ss-button ss-button--dark ss-button--base text-bold" data-id="1614839174935">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 512 512">
                      <path fill="currentColor" d="M93.95 290.03A327.038 327.038 0 0 0 .17 485.11l-.03.23c-1.7 15.28 11.21 28.2 26.49 26.51a327.02 327.02 0 0 0 195.34-93.8l75.4-75.4-128.02-128.02-75.4 75.4zM485.49 26.51c-35.35-35.35-92.67-35.35-128.02 0l-21.76 21.76-36.56-36.55c-15.62-15.62-40.95-15.62-56.56 0L138.47 115.84c-6.25 6.25-6.25 16.38 0 22.63l22.62 22.62c6.25 6.25 16.38 6.25 22.63 0l87.15-87.15 19.59 19.59L191.98 192 320 320.02l165.49-165.49c35.35-35.35 35.35-92.66 0-128.02z"></path>
                    </svg></div>
                  <div class="ss-button--label has-icon">Mark as responded</div>
                </div>
              </button><button class="ss-button ss-button--dark-inverse ss-button--base text-bold" data-id="1614839174935">
                <div class="ss-button--wrapper">
                  <div class="ss-button--icon"><svg xmlns="http://www.w3.org/2000/svg" class="" viewBox="0 0 384 512">
                      <path fill="currentColor" d="M0 512V48C0 21.49 21.49 0 48 0h288c26.51 0 48 21.49 48 48v464L192 400 0 512z"></path>
                    </svg></div>
                  <div class="ss-button--label has-icon">Mark as read</div>
                </div>
              </button></div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
  <div class="ss-pagination">
    <div class="ss-pagination--info">
      <p> Showing <span>1</span> to <span>10</span> of <span>20</span> results </p>
    </div>
    <div class="ss-pagination--page">
      <!--v-if--><span class="page-number active">1</span><span class="page-number">2</span><span class="nav-btn next"><img svg-inline=""></span>
    </div>
  </div>
</div>
`;
